### DISCUSSION TITLE
Updating PnP modern search v4 with future releases

### DISCUSSION THREAD

**Discussion by MrWorkingCat** [microsoft-search/pnp-modern-search#1913] [Q&A]
I am new to custom web parts and would like to know after I have deployed a version of the PnP web part to a site catalogue, how does the process to keep the package updated with the latest releases work?

Some of the questions I hope to answer are:

- Do I need to uninstall the current deployed version before upload a new version? Or can I upload the new version along side the previous one and keep both enabled?
- If I need to uninstall previous one, what happen with the web parts already added to SharePoint sites pages based the uninstalled version?
- Is  backwards compatibility a goal for the developers in future releases of version 4? If not, can we end up in scenarios where we need to rebuild the custom search pages from scratch?

**Comment by wobba** [comment]
You should be able to re-deploy over the existing sppkg and everything will continue to work as backward combability is important. If we do a breaking change we'll update that in the version number and text to make it clear. There should be no need to remove/re-add web parts on any page as they all reference the same id's always.

That said, if you have enabled CDN there are known issues that it takes some time for the CDN to kick in leading to errors on pages where the web parts are used. Because of this I recommend updating after hours or on the weekends to avoid SPFx caching issues.
